app/models/consumer_app.rb

```diff
@@ -2,12 +2,12 @@ class ConsumerApp < ApplicationRecord
   resourcify
 
   FOREM_BUNDLE = "com.forem.app".freeze
-  SUPPORTED_PLATFORMS = [Device::ANDROID, Device::IOS].freeze
-  FOREM_APP_PLATFORMS = [Device::IOS].freeze
+  FOREM_APP_PLATFORMS = %w[ios].freeze
 
-  validates :app_bundle, presence: true
-  validates :platform, presence: true
-  validates :app_bundle, uniqueness: { scope: :platform }
+  enum platform: { android: Device::ANDROID, ios: Device::IOS }
+
+  validates :app_bundle, presence: true, uniqueness: { scope: :platform }
+  validates :platform, inclusion: { in: platforms.keys }
```

Comment:
The DB stores `platform` as `iOS` or `Android` (with this case), so to be able to take advantage of ActiveRecord enums, I mapped `:ios` to `iOS` and `:android` to `Android`.

This way the code will use the symbols/strings in lower case and the DB won't have to change

Reply:
I gave this a try by destroying all Consumer Apps + Devices and starting fresh and the platform seems to be stored as all lowercased

<img width="790" alt="Screen Shot 2021-05-05 at 22 59 54" src="https://user-images.githubusercontent.com/6045239/117244198-d9f16200-adf5-11eb-8b20-689690ec52fd.png">

I'm not 100% sure, but maybe this would cause an inconsistency with records already existing in Forem instances. I only have access to BHC but those records are kept with each platform's "brand cased" (`iOS` / `Android`)

<img width="422" alt="Screen Shot 2021-05-05 at 23 00 23" src="https://user-images.githubusercontent.com/6045239/117244554-7b78b380-adf6-11eb-9ca6-c0c03fce9068.png">

I tried an experiment locally to try and keep the custom "brand case" and changed the enums to

```ruby
enum platform: { Android: Device::ANDROID, iOS: Device::IOS }
```

However, I realized this causes the helper methods to be "brand cased" and all the occurrences will need to change from things like `if consumer_app.android?` or `elsif consumer_app.ios?` to `if consumer_app.Android?` or `elsif consumer_app.iOS?`.

This doesn't feel too "natural" for Ruby/Rails to have anything other than snake_case for these though. What do you think about this situation? Maybe we can add a DataUpdateScript here to migrate all occurrences to the lowercased version and keep the enum all lowercase?

Reply:
Hi @fdoxyz thanks for the test!

Rails will show you the value lowercased, but if you check the DB itself you'll notice how they are stored with `iOS` and `Android`, not to break compatibility.

```sql
> psql Forem_development
Forem_development=# select id, platform from consumer_apps;
 id | platform
----+----------
  1 | iOS
  4 | Android
(2 rows)
```

that's why we're using enums, to keep the data as it is but map it with something handier at the Ruby level

Reply:
Oh I see, the console results misled me thanks for clarifying. That's perfect then!